### PR TITLE
Improve log message for ORC and table definition mismatch

### LINKED ISSUE #12778: Improve the log message when there are mismatches between ORC and table definition
This is the current log message when there are mismatches between ORC and table definition, the `.` in `.id` is annoying and may mislead people, also the format of data type is not consistent, one is in lower case and the other is in upper case
```
Malformed ORC file. Can not read SQL type bigint from ORC stream .id of type STRING
```

### PR DESCRIPTION
**Pull request by ArvinZheng**
Fix #12778

### CODE REVIEW

File: lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java [hunk 1]
```diff
@@ -334,7 +334,14 @@ private static OrcColumn createOrcColumn(
             ColumnMetadata<OrcType> types,
             OrcDataSourceId orcDataSourceId)
     {
-        String path = fieldName.isEmpty() ? parentStreamName : parentStreamName + "." + fieldName;
+        final String path;
```
Comment (electrum) [inline lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java:337]: Nit: we don't use `final` for local variables

File: lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java [hunk 2]
```diff
@@ -334,7 +334,14 @@ private static OrcColumn createOrcColumn(
...
+        if (fieldName.isEmpty()) {
+            path = parentStreamName;
+        }
+        // parentStreamName can be empty
```
Comment (electrum) [inline lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java:341]: No need for the comment, since otherwise we wouldn't do the `isEmpty()` check

**Comment by ArvinZheng** [comment]
@electrum , addressed all comments and tests are passed, pls have another look when you get a chance, thank you.

**Comment by ArvinZheng** [comment]
@electrum mind taking another look?

**Comment by mosabua** [comment]
:wave: @ArvinZheng - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by ArvinZheng** [comment]
@mosabua just rebased the branch

**Comment by mosabua** [comment]
Thank you for rebasing. This looks good to me. @electrum or @martint could you have a look?

**Pull request closed without merging**